src/LayerPlan.cpp

```diff
@@ -74,52 +74,66 @@ GCodePath* GCodePlanner::getLatestPathWithConfig(GCodePathConfig* config, SpaceF
     return ret;
 }
 
-void GCodePlanner::forceNewPathStart()
+void LayerPlan::forceNewPathStart()
 {
     std::vector<GCodePath>& paths = extruder_plans.back().paths;
     if (paths.size() > 0)
         paths[paths.size()-1].done = true;
 }
 
-GCodePlanner::GCodePlanner(SliceDataStorage& storage, int layer_nr, int z, int layer_thickness, Point last_position, int current_extruder, bool is_inside_mesh, std::vector<FanSpeedLayerTimeSettings>& fan_speed_layer_time_settings_per_extruder, CombingMode combing_mode, int64_t comb_boundary_offset, bool travel_avoid_other_parts, int64_t travel_avoid_distance)
+LayerPlan::LayerPlan(const SliceDataStorage& storage, int layer_nr, int z, int layer_thickness, unsigned int start_extruder, const std::vector<FanSpeedLayerTimeSettings>& fan_speed_layer_time_settings_per_extruder, CombingMode combing_mode, int64_t comb_boundary_offset, bool travel_avoid_other_parts, int64_t travel_avoid_distance)
```

Comment:
There are a couple of places where factories could/should be used.
There are a couple of functions which have multiple parameters with defaults, where you would have to reiterate the defaults if you only want a special value for the 4th parameter with default.

I don't think this is one of them, though.
The constructor is basically only used once in processLayer and there all parameters need to be filled in.
It's then also used in processRaft, but I don't think we should introduce default values for when it is called for processing raft, because the main purpose of the constructor is not for usage in raft.

I think this constructor is only used 4 times and there should be no default values, so I don't think there's enough reason to begin a factory.

In other places using factories is long overdue ':{